spec/models/index_manager_spec.rb

```diff
@@ -0,0 +1,181 @@
+      # This incremental is before the full dump, don't run it
+      full_event = FactoryBot.create(:full_dump_event, start: Time.current - 1.day, finish: Time.current - 1.day + 100)
+
+      index_manager = described_class.for(solr_url)
+      Sidekiq::Testing.inline! do
+        # Index full dump
+        index_manager.index_next_dump!
+        run_callback(Sidekiq::BatchSet.new.to_a.last)
+        # Index incremental
+      end
+
+      index_manager = described_class.find(index_manager.id)
+      expect(index_manager.index_next_dump!).to be_nil
+      expect(index_manager.reload.dump_in_progress).to be_nil
+      expect(DumpFileIndexJob).to have_received(:perform_async).exactly(2).times
+      expect(index_manager).not_to be_in_progress
+    end
+    it "indexes the previous incremental if the most recent full dump has been done" do
+      allow(DumpFileIndexJob).to receive(:perform_async).and_call_original
+      # This incremental is before the pre-full-dump incremental, don't run it
+      pre_pre_incremental_event = FactoryBot.create(:incremental_dump_event, start: Time.current - 2.days, finish: Time.current - 2.days + 100)
+      pre_incremental_event = FactoryBot.create(:incremental_dump_event, start: Time.current - 2.days, finish: Time.current - 2.days + 100)
+      full_event = FactoryBot.create(:full_dump_event, start: Time.current - 1.day, finish: Time.current - 1.day + 100)
+      # This should get run on the third call
+      incremental_event = FactoryBot.create(:incremental_dump_event, start: Time.current - 4.hours, finish: Time.current - 3.hours)
+      # Incremental that isn't run yet, but would eventually.
+      final_incremental_event = FactoryBot.create(:incremental_dump_event, start: Time.current - 2.hours, finish: Time.current - 1.hour)
+
+      index_manager = described_class.for(solr_url)
+      Sidekiq::Testing.inline! do
+        # Index full dump
+        index_manager.index_next_dump!
+        run_callback(Sidekiq::BatchSet.new.to_a.last)
+        # Index pre-full incremental
+        index_manager = described_class.find(index_manager.id)
+        index_manager.index_next_dump!
+        run_callback(Sidekiq::BatchSet.new.to_a.last)
+        # Index post-full incremental
+        index_manager = described_class.find(index_manager.id)
+        index_manager.index_next_dump!
+        run_callback(Sidekiq::BatchSet.new.to_a.last)
+      end
+      solr.commit
+
+      expect(DumpFileIndexJob).not_to have_received(:perform_async).with(pre_pre_incremental_event.dump.dump_files.first.id, anything)
+      expect(DumpFileIndexJob).to have_received(:perform_async).with(pre_incremental_event.dump.dump_files.first.id, anything)
+      expect(DumpFileIndexJob).to have_received(:perform_async).with(incremental_event.dump.dump_files.first.id, anything)
+      expect(DumpFileIndexJob).not_to have_received(:perform_async).with(final_incremental_event.dump.dump_files.first.id, anything)
```

Comment:
This one doesn't run just because you didn't run the callbacks manually enough times?

Reply:
Yes.